File: ImageResize/ImageResize/ViewController.m
//
//  ViewController.m
//  ImageResize
//
//  Created by kuroky on 2019/5/21.
//  Copyright © 2019 Emucoo. All rights reserved.
//

#import "ViewController.h"
#import "ImageViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property (nonatomic, strong) NSArray *dataList;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationItem.title = @"Resize";
    self.dataList = @[@"Core Graphics", @"CoreImage", @"ImageIO", @"UIKit", @"vImage", @"ImageIO1"];
    
    self.tableView.dataSource = self;
    self.tableView.delegate = self;
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = self.dataList[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    ImageViewController *imageView = [sb instantiateViewControllerWithIdentifier:@"ImageViewController"];
    imageView.resizeType = indexPath.row + 1;
    [self.navigationController pushViewController:imageView animated:YES];
}

@end

File: Image-Resizing-Example-master/Example/ViewController.swift
import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var tableView: UITableView!
    
    let dataList = ["Core Graphics", "CoreImage", "ImageIO", "UIKit", "vImage", "ImageIO1"]
    
    override func viewDidLoad() {
        self.navigationItem.title = "Resize"
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
        self.tableView.rowHeight = 60
    }
}

extension ViewController: UITableViewDataSource, UITableViewDelegate {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        cell.textLabel?.text = self.dataList[indexPath.row]
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        let sb = UIStoryboard.init(name: "Main", bundle: nil)
        let imageViewController = sb.instantiateViewController(withIdentifier: "ImageViewController")  as! ImageViewController
        imageViewController.resizeType = indexPath.row + 1
        self.navigationController?.pushViewController(imageViewController, animated: true)
    }
}
